text/2019-02-26-tikv-client-command-line.md

```diff
@@ -0,0 +1,232 @@
+```
+
+On the command line, these options have the same name, but with `_` replaced
+with `-` for idiomatic reasons.
+
+### Input/Output
+
+By default input is by positional arguments in `bash`/`sh` string format.
+
+By default the tool will output in prettyprinted JSON. For single entry results,
+it will return a string (including `"`), for key-value pairs it will use a JSON
+object `{ key: "", value: "" }`, for multiple results it will return a JSON
+array. Errors will use an `{ error: "" }` object. Opting out of prettyprint will
+be via the `--minify` flag.
+
+JSON is chosen because it is an efficient, pervasive, human-readable, modern
+standard that has many tools (eg `jq`) to use for post processing if needed.
+
+All standard (non-logging) output will be on `stdout`.
+
+It will feature configurable logging levels with `slog` which will output to `stderr`.
+
+The tool will exit with exit code zero on successful requests, and non-zero on unsuccessful
+requests. In the case of REPL mode, it shall exit zero unless the connection is
+found to be lost.
+
+### Interacting with it
+
+Users will interact with the tool through a command line interface. It will
+support both one-off mode or REPL (Read-eval-print-loop) style mode (See [alternatives](#Alternatives)).
+
+TiKV has two APIs, *raw* and *transactional*, both are supported by this tool.
+Learn about the differences [here](https://tikv.org/docs/architecture/#apis).
+
+Example of the one-off raw mode:
+
+```bash
+$ tikv-cli --mode raw set "cncf" "linux foundation"
+Finished in 0.001s.
+$ tikv-cli --mode raw set "tikv" "rust"
+Finished in 0.001s.
+$ tikv-cli --mode raw get "cncf"
+"linux foundation"
+Finished in 0.002s.
+$ tikv-cli --mode raw scan "b"..
+[
+    { key: "cncf", value: "linux foundation" },
+    { key: "tikv", value: "rust" },
+]
+Finished in 0.003s.
+```
+
+Example of REPL raw mode:
+
+```bash
+$ tikv-cli --mode raw
+> set "cncf" "linux foundation"
+Finished in 0.001s.
+> set "tikv" "rust"
+Finished in 0.001s.
+> get "cncf"
+"linux foundation"
+Finished in 0.001s.
+> scan "b"..
+[
+    { key: "cncf", value: "linux foundation" },
+    { key: "tikv", value: "rust" },
+]
+Finished in 0.003s.
+```
+
+In the REPL of the *(default)* transactional mode below, the `>>` communicates
+that the user is in a transaction. Also notice how the transaction timestamp is
+outputted in the request complete messages.
+
+```bash
+$ tikv-cli
+> begin
+Finished in 0.000s. (txn ts: 1551216224838)
+>>set "cncf" "linux foundation"
+Finished in 0.001s. (txn ts: 1551216224838)
+>> set "tikv" "rust"
+Finished in 0.001s. (txn ts: 1551216224838)
+>> get "cncf"
+"linux foundation"
+Finished in 0.001s. (txn ts: 1551216224838)
+>> scan "b"..
+[
+    { key: "cncf", value: "linux foundation" },
+    { key: "tikv", value: "rust" },
+]
+Finished in 0.003s. (txn ts: 1551216224838)
+>> commit
+Finished in 0.001s. (txn ts: 1551216224838)
+```
+
+One-off transactional mode is a bit different, a series of commands can be

```

Comment:
I think `Auto-commit mode` will be more understandable for me ^^

Reply:
I clarified the description and added that term :)